src/main/java/com/parsing/pdf/parsing/DataRecognizer.java

```diff
@@ -0,0 +1,133 @@
+package com.parsing.pdf.parsing;
+
+import com.parsing.pdf.parsing.model.Column;
+import com.parsing.pdf.parsing.model.Row;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class DataRecognizer {
+
+    private final LotPDFResultService lotPDFResultService;
+    private String model = null;
+    private Integer amount = 0;
+    private BigDecimal price = null;
+    private BigDecimal totalPrice = null;
+
+    @Value("${laptop.models}")
+    private final List<String> laptopModels;
+
+    public final void recognizeLotPDFResult(List<Row> rows) {
```

Comment:
Consider dividing this method for couple of methods, it looks too big and too complicated.

Reply:
I have splitted for several methods.